src/main/java/org/folio/fqm/service/FqlToSqlConverterService.java

```diff
@@ -190,22 +190,22 @@ private static Condition handleDate(FieldCondition<?> fieldCondition, org.jooq.F
 
     LocalDateTime nextDayDateTime = dateTime.plusDays(1);
     if (fieldCondition instanceof EqualsCondition) {
-      condition = field.greaterOrEqual(dateTime.format(DATE_TIME_FORMATTER))
-        .and(field.lessThan(nextDayDateTime.format(DATE_TIME_FORMATTER)));
+      condition = field.greaterOrEqual(valueField(dateTime.format(DATE_TIME_FORMATTER), fieldCondition, entityType))
+        .and(field.lessThan(valueField(nextDayDateTime.format(DATE_TIME_FORMATTER), fieldCondition, entityType)));
     } else if (fieldCondition instanceof NotEqualsCondition) {
-      condition = field.greaterOrEqual(nextDayDateTime.format(DATE_TIME_FORMATTER))
-        .or(field.lessThan(dateTime.format(DATE_TIME_FORMATTER)));
+      condition = field.greaterOrEqual(valueField(nextDayDateTime.format(DATE_TIME_FORMATTER), fieldCondition, entityType))
+        .or(field.lessThan(valueField(dateTime.format(DATE_TIME_FORMATTER), fieldCondition, entityType)));
     } else if (fieldCondition instanceof GreaterThanCondition greaterThanCondition) {
       if (greaterThanCondition.orEqualTo()) {
-        condition = field.greaterOrEqual(dateTime.format(DATE_TIME_FORMATTER));
+        condition = field.greaterOrEqual(valueField(dateTime.format(DATE_TIME_FORMATTER), fieldCondition, entityType));
       } else {
-        condition = field.greaterOrEqual(nextDayDateTime.format(DATE_TIME_FORMATTER));
+        condition = field.greaterOrEqual(valueField(nextDayDateTime.format(DATE_TIME_FORMATTER), fieldCondition, entityType));
       }
     } else if (fieldCondition instanceof LessThanCondition lessThanCondition) {
       if (lessThanCondition.orEqualTo()) {
-        condition = field.lessThan(nextDayDateTime.format(DATE_TIME_FORMATTER));
+        condition = field.lessThan(valueField(nextDayDateTime.format(DATE_TIME_FORMATTER), fieldCondition, entityType));
       } else {
-        condition = field.lessThan(dateTime.format(DATE_TIME_FORMATTER));
+        condition = field.lessThan(valueField(dateTime.format(DATE_TIME_FORMATTER), fieldCondition, entityType));
```

Comment:
This is the change we talked about in chat, about running all of the date values through `valueField()` so that it can use the `valueFunction`